xds/src/main/java/io/grpc/xds/XdsClientImpl.java

```diff
@@ -414,6 +427,39 @@ void cancelEndpointDataWatch(String clusterName, EndpointWatcher watcher) {
     }
   }
 
+  @Override
+  void watchListenerData(int port, ListenerWatcher watcher) {
+    checkState(configWatcher == null, "ConfigWatcher for %s already registered", hostName);
+    checkState(listenerWatcher == null, "ListenerWatcher already registered");
+    listenerWatcher = checkNotNull(watcher, "watcher");
+    checkState(port > 0, "port needs to be > 0");
+    this.port = port;
+    logger.log(XdsLogLevel.INFO, "Started watching listener for port {0}", port);
+    if (rpcRetryTimer != null && rpcRetryTimer.isPending()) {
+      // Currently in retry backoff.
+      return;
+    }
+    if (adsStream == null) {
+      startRpcStream();
+    }
+    updateNodeMetadataForListenerRequest(port);
+    adsStream.sendXdsRequest(ADS_TYPE_URL_LDS, null);
```

Comment:
Sending an empty list vs null has different implications on the xDS server side and the xDS folks (TD) told me they expect null in this case.

Reply:
I found this the hard way while testing with the xDS server (TD). An uninitialized list field and initialized list of 0 size are different apparently. I'll post a pointer if I find one.

Reply:
As far as I know, calling `addAllXXX()` method with an empty list is equivalent to not setting that field in protobuf. There isn't a real `null` semantics in protos. Empty collections or strings are not sent out in wire and getters return at least an empty collection even if the field is not set.

Reply:
I examine the code. addAll(emptyList) has side effect of setting the internal collection to `ArrayList` instead of `Collections.emptyList()`. usually emptyList can cause more problems...

Reply:
But it shouldn't have behavior-wise difference than not setting it, right? Anyway, @sanjaypujare said the design is still half baked and later it will contain some resources explicitly.

Reply:
It's not about the design being half-baked or half-ready. It is about how the server treats the request differently: I didn't pursue why they don't treat both the same but I need to do what the server expects.

Reply:
Changed to using `Collections.emptyList()`